native/packages/react-native-bpk-component-progress/src/BkpProgress.js

```diff
@@ -0,0 +1,129 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyleSheet, View, Animated, Easing } from 'react-native';
+import {
+  animmationDurationBase,
+  spacingMd,
+} from 'bpk-tokens/tokens/base.react.native';
+
+import STYLES from './styles';
+
+const styles = StyleSheet.create({
+  container: {
+    height: spacingMd,
+    overflow: 'hidden',
+  },
+  fill: {
+    height: spacingMd,
+  },
+});
+
+type Props = {
+  max: number,
+  min: number,
+  value: number,
+  type: 'Default' | 'Bar',
+  trackStyle: StyleSheet,
```

Comment:
Does `Stylesheet` work (it should be optional btw)? We've ended up using `?any` in the past because anything we tried was either wrong or didn't provide value...

Reply:
Strictly speaking what you pass for the `style` is a property of a `StyleSheet` so it isn't correct even if it does work. Let's stick with `?any` and revisist it at later date

Reply:
Haven't tried to provide the style yet, guess I'll find out in the tests :)

Reply:
I tried with the measure api but I haven't found a way to do this without state because I need the width for the interpolate function. Even if I use the measure api I need to use state to trigger a rerender after getting the real width. The good things is, I tested and onLayout is only trigger after the first mount, or if the orientation/style changes. Other updates don't trigger it again.